nat-lab/tests/test_reconnections.py

```diff
@@ -16,6 +16,7 @@
 
 @pytest.mark.asyncio
 @pytest.mark.derp
+@pytest.mark.xfail(reason="test is flaky - Jira issue: LLT-4161")
```

Comment:
Not sure about this test

Reply:
If is flaky?

Reply:
yes

Reply:
Hmm, if I found which pipeline it was, it looks like some random fail, the logs look ok.

Reply:
@stalowyjez it's on the linked Jira issue: https://bucket.digitalarsenal.net/low-level-hacks/vpn/client/libtelio-build/-/jobs/8645474

Reply:
On these logs above, the expected behaviour is printed to stdout but it timesout when checking for the events. 

On this more recent pipeline the test failed again: https://bucket.digitalarsenal.net/low-level-hacks/vpn/client/libtelio-build/-/jobs/8682968#L2913

It failed also with different containers/params but the failure is the same: `wait_for_event_on_any_derp` timesout, even though the derp connection is logged to stdout. So my suspicion is the usual race condition between the action and `wait_for_event` functions. It can be reproduced introducing a `asyncio.sleep(1)` after `set_meshmap()`. I think one fix could be launching `wait_for_event` tasks asynchronously before restarting the device, or checking for the state instead of event.. I think the former fix suits better as the `wait_for_state()` function only checks the last reported state.

Reply:
@gytsto this was the reason for including an `asyncio.Event()` as parameter of `wait_for_event()` functions, to guarantee we are waiting for the event before doing the action (restarting device in this case).

Reply:
Ok, let's leave it as it is for now